src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -273,6 +280,18 @@ private Task ExecuteWithinLockAsync(Func<Task> task)
             }).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Roslyn is working on supporting free-threaded initialization.
+        /// in order to support a smooth transition, we decide whether to serialize
+        /// based on whether a particular member that was added at the same time exists.
+        /// </summary>
+        /// <returns><see langword="true"/> if the workspace supports batching.</returns>
+        internal static bool WorkspaceSupportsBatchingAndFreeThreadedInitialization()
+        {
+            return typeof(IWorkspaceProjectContext).GetMethod("StartBatch") != null &&
```

Comment:
Is it worth caching this? Not sure how expensive this is for repeated lookups.

Reply:
Removed in latest.

Reply:
Cached in latest.  Not sure whether worth it or not either.